test/e2e/specs/editor/various/writing-flow.spec.js

```diff
@@ -0,0 +1,1004 @@
+			.toBe( 'core/paragraph' );
+		// Arrow left back to Column block.
+		await page.keyboard.press( 'ArrowLeft' );
+		await expect
+			.poll( writingFlowUtils.getActiveBlockName )
+			.toBe( 'core/column' );
+		// Arrow left back to Columns block.
+		await page.keyboard.press( 'ArrowLeft' );
+		await expect
+			.poll( writingFlowUtils.getActiveBlockName )
+			.toBe( 'core/columns' );
+		// Arrow up to first paragraph.
+		await page.keyboard.press( 'ArrowUp' );
+		await expect
+			.poll( writingFlowUtils.getActiveBlockName )
+			.toBe( 'core/paragraph' );
+	} );
+
+	test( 'should navigate around inline boundaries', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		// Add demo content.
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'First' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'Second' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'Third' );
+
+		// Navigate to second paragraph.
+		await pageUtils.pressKeyTimes( 'ArrowLeft', 6 );
+
+		// Bold second paragraph text.
+		await page.keyboard.down( 'Shift' );
+		await pageUtils.pressKeyTimes( 'ArrowLeft', 6 );
+		await page.keyboard.up( 'Shift' );
+		await pageUtils.pressKeyWithModifier( 'primary', 'b' );
+
+		// Arrow left from selected bold should collapse to before the inline
+		// boundary. Arrow once more to traverse into first paragraph.
+		await page.keyboard.press( 'ArrowLeft' );
+		await page.keyboard.press( 'ArrowLeft' );
+		await page.keyboard.type( 'After' );
+
+		// Arrow right from end of first should traverse to second, *BEFORE*
+		// the bolded text. Another press should move within inline boundary.
+		await pageUtils.pressKeyTimes( 'ArrowRight', 2 );
+		await page.keyboard.type( 'Inside' );
+
+		// Arrow left from end of beginning of inline boundary should move to
+		// the outside of the inline boundary.
+		await pageUtils.pressKeyTimes( 'ArrowLeft', 6 );
+		await page.keyboard.press( 'ArrowLeft' ); // Separate for emphasis.
+		await page.keyboard.type( 'Before' );
+
+		// Likewise, test at the end of the inline boundary for same effect.
+		await page.keyboard.press( 'ArrowRight' ); // Move inside
+		await pageUtils.pressKeyTimes( 'ArrowRight', 12 );
+		await page.keyboard.type( 'Inside' );
+		await page.keyboard.press( 'ArrowRight' );
+
+		// Edge case: Verify that workaround to test for ZWSP at beginning of
+		// focus node does not take effect when on the right edge of inline
+		// boundary (thus preventing traversing to the next block by arrow).
+		await page.keyboard.press( 'ArrowRight' );
+		await page.keyboard.press( 'ArrowLeft' );
+
+		// Should be after the inline boundary again.
+		await page.keyboard.type( 'After' );
+
+		// Finally, ensure that ArrowRight from end of unbolded text moves to
+		// the last paragraph.
+		await page.keyboard.press( 'ArrowRight' );
+		await page.keyboard.type( 'Before' );
+
+		expect( await editor.getEditedPostContent() ).toMatchSnapshot();
```

Comment:
We might want to also use `expect.poll` here for consistency?

Reply:
I forgot that  `expect.poll()` does not support `toMatchSnapshot` matcher; this was the reason for the inconsistency.

Reply:
Huh, TIL! Thanks for double-checking!